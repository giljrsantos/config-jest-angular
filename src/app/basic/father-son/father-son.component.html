<h5>Filho</h5>
<code>{{ client | json }}</code>

<br /><br />

<button
  *ngIf="client"
  class="btn btn-outline-primary"
  (click)="onChange(5)"
  data-test="btn-id"
>
  Mudar ID
</button>
<button
  *ngIf="client"
  class="btn btn-outline-danger"
  (click)="onDelete()"
  data-test="btn-delete"
>
  Apagar Cliente
</button>
