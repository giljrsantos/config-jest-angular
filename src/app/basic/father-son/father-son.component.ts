import {
  Component,
  EventEmitter,
  Input,
  Output,
} from '@angular/core';

import { Client } from '../interface';

@Component({
  selector: 'app-father-son',
  templateUrl: './father-son.component.html',
  styleUrls: ['./father-son.component.scss'],
})
export class FatherSonComponent {
  @Input() client?: Client;
  @Output() onDeleteClient = new EventEmitter();
  @Output() onClientUpdated = new EventEmitter<Client>();

  onDelete() {
    this.client = undefined;
    this.onDeleteClient.emit();
  }

  onChange(id: number) {
    if (!this.client) return;
    //this.client.id = newId; n√£o usar
    this.client = {
      ...this.client,
      id,
    };
    this.onClientUpdated.emit({ ...this.client });
  }
}
