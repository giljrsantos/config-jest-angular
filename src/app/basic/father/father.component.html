<h1>Pai</h1>
<button
  class="btn btn-primary"
  (click)="onSetClient('Gilberto')"
>
  Carregar Cliente
</button>
<br /><br />

<code class="mt-2">
  {{ client | json }}
</code>

<hr />

<app-father-son
  [client]="client"
  (onDeleteClient)="client = undefined"
  (onClientUpdated)="client = $event"
/>
