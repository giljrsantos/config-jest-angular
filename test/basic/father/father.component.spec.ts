import { ComponentFixture, TestBed } from '@angular/core/testing';

import { By } from '@angular/platform-browser';
import { FatherComponent } from '../../../src/app/basic/father/father.component';
import { FatherSonComponent } from '../../../src/app/basic/father-son/father-son.component';

describe('FatherComponent', () => {
  let component: FatherComponent;
  let fixture: ComponentFixture<FatherComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [FatherComponent, FatherSonComponent]
    }).compileComponents()
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(FatherComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
    jest.clearAllMocks();
  });
  it('Deve Criar', () => {
    expect(component).toBeTruthy();
  });

  it('deve corresponder ao snpashot', () => {
    expect(compiled).toMatchSnapshot();
  });

  it('deve estabelecer o cliente com o nome indicado', () => {
    component.onSetClient('Pedro');
    fixture.detectChanges();

    const codeDiv = compiled.querySelector('.mt-2');

    expect(codeDiv?.textContent).toContain('"name"');
    expect(codeDiv?.textContent).toContain('"Pedro"');
  });

  it('deve excluir o cliente se ele for emitido onDeleteClient (Filho)', () => {
    component.client = { id: 1, name: 'Eduardo' }
    fixture.detectChanges();

    const sonDebugElement = fixture.debugElement.query(By.directive(FatherSonComponent));
    const sonComponent: FatherSonComponent = sonDebugElement.componentInstance;

    sonComponent.onDeleteClient.emit();
    expect(component.client).toBe(undefined);
  });

  it('deve atualizar o cliente onClientUpdated', () => {
    component.client = { id: 10, name: 'Eduardo' };
    fixture.detectChanges();

    const sonDebugElement = fixture.debugElement.query(By.directive(FatherSonComponent));
    const sonComponet: FatherSonComponent = sonDebugElement.componentInstance;

    sonComponet.onClientUpdated.emit({ id: 10, name: 'Pedro' });
    expect(component.client).toEqual({ id: 10, name: 'Pedro' });
  });

});
