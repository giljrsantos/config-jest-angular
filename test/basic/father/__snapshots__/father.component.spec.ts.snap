// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FatherComponent deve corresponder ao snpashot 1`] = `
<div
  id="root1"
>
  <h1>
    Pai
  </h1>
  <button
    class="btn btn-primary"
  >
     Carregar Cliente

  </button>
  <br />
  <br />
  <code
    class="mt-2"
  >
     

  </code>
  <hr />
  <app-father-son>
    <h5>
      Filho
    </h5>
    <code>
      
    </code>
    <br />
    <br />
    
    
  </app-father-son>
</div>
`;
